### PR TITLE
Fix deep merge for existing `env` on work-pools

### PR DESCRIPTION
**Pull request by zzstoatzz**
currently on `main`, we correctly deep merge `env` like `deployment | flow_run`, but this merged result totally overwrites any hardcoded `env` defined directly on the work pool.

this PR makes sure we deep merge the result of `deployment | flow_run` with any `env` that may be defined on the work pool itself

### CODE REVIEW

File: src/prefect/server/models/workers.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ async def update_work_pool(
             [UUID, pendulum.DateTime, orm_models.WorkPool, orm_models.WorkPool],
             Awaitable[None],
         ]
-    ],
+    ] = None,
```
Comment (cicdw) [inline src/prefect/server/models/workers.py:191]: this is small but i love it when writing a test uncovers some mild UX edge case like this

File: tests/workers/test_base_worker.py [hunk 2]
```diff
@@ -1803,3 +1806,113 @@ async def test_env_merge_logic_is_deep(prefect_client, session, flow):
...
+            entrypoint="/file.py:flow",
+            parameter_openapi_schema={},
+        ),
+    )
+    await session.commit()
+
+    flow_run = await prefect_client.create_flow_run_from_deployment(
+        deployment.id,
+        state=Pending(),
+        job_variables={"env": {"C": "3", "D": "4"}},
```
Comment (cicdw) [inline tests/workers/test_base_worker.py:1908]: for this test would you mind putting `"C"` on the deployment itself just to make sure the three all got merged?

**Review by cicdw — CHANGES_REQUESTED** [review]
small request for a test tweak, otherwise this LGTM

File: tests/workers/test_base_worker.py (hunk 2 above)
Comment (zzstoatzz) [reply tests/workers/test_base_worker.py:1909]: good call! added in [01b7bc5](https://github.com/PrefectHQ/prefect/pull/15325/commits/01b7bc57ed5ab9ab63faa4b0194198b5605ebeb0)

File: tests/workers/test_base_worker.py [hunk 3]
```diff
@@ -1774,7 +1775,67 @@ def create_run_with_deployment(state):
...
+            {"A": "1", "B": "2"},
+        ),
+    ],
+    ids=[
+        "flow_run_into_deployment",
+        "deployment_into_work_pool",
+        "flow_run_into_work_pool",
+        "try_overwrite_with_empty_str",
+    ],
+)
```
Comment (desertaxle) [inline tests/workers/test_base_worker.py:1812]: I like this.

**Review by desertaxle — APPROVED** [review]
